builder/module-options.nix

```diff
@@ -0,0 +1,272 @@
+{ config, lib, ... }:
+# TODO: add support for assertions
+let
+  latestRelease = import ../latest-release.nix;
+
+  releaseHashes = import ../hashes/${config.release}.nix;
+  hashedTarget = releaseHashes.targets.${config.hardware.target}.${config.hardware.subtarget};
+  hashedFeeds = releaseHashes.packages.${config.hardware.arch};
+
+  defaultSumsFileSha256 = hashedTarget.sha256;
+
+  # TODO: get rid of `pkgs`
+  profiles = import ../profiles.nix { pkgs = null; inherit (config) release; };
+  detectedProfile = profiles.identifyProfile config.hardware.profile;
+  # TODO: move arch to profiles
+  detectedArch = hashedTarget.packagesArch;
```

Comment:
`arch` is already a property in `profiles.json`. I feel like it's more fitting to get this info from there, than from the feeds, since it's a property of a specific board/target